app/code/Magento/Inventory/Plugin/SourceTypeAttribute.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Plugin;
+
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\Inventory\Model\SourceTypeLinkManagement;
+
+class SourceTypeAttribute
+{
+    /**
+     * @var SourceTypeLinkManagement
+     */
+    protected $sourceTypeLinkManagement;
+
+    /**
+     * SourceTypeAttribute constructor.
+     * @param SourceTypeLinkManagement $sourceTypeLinkManagement
+     */
+    public function __construct(
+        SourceTypeLinkManagement $sourceTypeLinkManagement
+    ) {
+        $this->sourceTypeLinkManagement = $sourceTypeLinkManagement;
+    }
+
+    /**
+     * @param SourceRepositoryInterface $subject
+     * @param SourceInterface $source
+     * @return SourceInterface
+     */
+    public function afterGet(
+        SourceRepositoryInterface $subject,
+        SourceInterface $source
+    ) {
+        $this->sourceTypeLinkManagement->loadTypeLinksBySource($source);
```

Comment:
What happens here is not absolutely clear.

It might be better to load type links and assign them to the source as extension attributes implicitly in the plugin